ee/hogai/artifacts/manager.py

```diff
@@ -22,6 +23,44 @@
 from ee.hogai.utils.types.base import ArtifactRefMessage, AssistantMessageUnion
 from ee.models.assistant import AgentArtifact
 
+Content = VisualizationArtifactContent | NotebookArtifactContent
+ContentT = TypeVar("ContentT", bound=Content)
```

Comment:
We have a type for the content in the schema:

```suggestion
ContentT = TypeVar("ContentT", bound=ArtifactContent)
```